src/catalyst-toolbox/catalyst-toolbox/src/archive/vit_ss.rs

```diff
@@ -0,0 +1,133 @@
+use csv::Writer;
+use serde::{Deserialize, Serialize};
+use std::{fs::File, path::Path};
+use vit_servicing_station_lib::{
+    db::{
+        self, load_db_connection_pool,
+        queries::{
+            funds::query_fund_by_id,
+            snapshot::{
+                query_contributions_by_snapshot_tag, query_snapshot_by_tag,
+                query_voters_by_snapshot_tag,
+            },
+        },
+    },
+    utils::serde::{deserialize_unix_timestamp_from_rfc3339, serialize_unix_timestamp_as_rfc3339},
+    v0::errors::HandleError,
+};
+
+#[derive(Debug, thiserror::Error)]
+pub enum Error {
+    #[error(transparent)]
+    DatabaseError(#[from] db::Error),
+
+    #[error(transparent)]
+    FetchError(#[from] HandleError),
+
+    #[error(transparent)]
+    Io(#[from] std::io::Error),
+
+    #[error(transparent)]
+    Csv(#[from] csv::Error),
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Eq)]
+pub struct Fund {
+    #[serde(default = "Default::default")]
+    pub id: i32,
+    #[serde(alias = "fundName")]
+    pub fund_name: String,
+    #[serde(alias = "fundGoal")]
+    pub fund_goal: String,
+    #[serde(alias = "votingPowerThreshold")]
+    pub voting_power_threshold: i64,
+    #[serde(alias = "fundStartTime")]
+    #[serde(deserialize_with = "deserialize_unix_timestamp_from_rfc3339")]
+    #[serde(serialize_with = "serialize_unix_timestamp_as_rfc3339")]
+    pub fund_start_time: i64,
+    #[serde(alias = "fundEndTime")]
+    #[serde(deserialize_with = "deserialize_unix_timestamp_from_rfc3339")]
+    #[serde(serialize_with = "serialize_unix_timestamp_as_rfc3339")]
+    pub fund_end_time: i64,
+    #[serde(alias = "nextFundStartTime")]
+    #[serde(deserialize_with = "deserialize_unix_timestamp_from_rfc3339")]
+    #[serde(serialize_with = "serialize_unix_timestamp_as_rfc3339")]
+    pub next_fund_start_time: i64,
+    #[serde(alias = "registrationSnapshotTime")]
+    #[serde(deserialize_with = "deserialize_unix_timestamp_from_rfc3339")]
+    #[serde(serialize_with = "serialize_unix_timestamp_as_rfc3339")]
+    pub registration_snapshot_time: i64,
+    #[serde(alias = "nextRegistrationSnapshotTime")]
+    #[serde(deserialize_with = "deserialize_unix_timestamp_from_rfc3339")]
+    #[serde(serialize_with = "serialize_unix_timestamp_as_rfc3339")]
+    pub next_registration_snapshot_time: i64,
+    // #[serde(flatten)]
+    // pub stage_dates: FundStageDates,
+    #[serde(alias = "resultsUrl")]
+    pub results_url: String,
+    #[serde(alias = "surveyUrl")]
+    pub survey_url: String,
+}
+
+fn csv_writer(output_dir: &Path, name: String) -> Result<Writer<File>, Error> {
+    let mut path = output_dir.to_path_buf();
+    path.push(name);
+    let file = std::fs::File::create(path).unwrap();
```

Comment:
This can return `Error` instead:

```suggestion
    let file = std::fs::File::create(path).map_err(Error::Io)?;
```

Reply:
fixed